aks-node-controller/pkg/nodeconfigutils/utils_test.go

```diff
@@ -0,0 +1,179 @@
+			got.Location, want.Location)
+	}
+}
+
+// validateApiServerConfig validates API server configuration fields.
+func validateApiServerConfig(t *testing.T, got, want *aksnodeconfigv1.ApiServerConfig) {
+	if want == nil || got == nil {
+		return
+	}
+
+	if got.ApiServerName != want.ApiServerName {
+		t.Errorf("UnmarshalConfigurationV1() apiServerName = %v, want %v",
+			got.ApiServerName, want.ApiServerName)
+	}
+}
+
+func TestUnmarshalConfigurationV1(t *testing.T) {
+	tests := []struct {
+		name    string
+		data    []byte
+		want    *aksnodeconfigv1.Configuration
+		wantErr bool
+	}{
+		{
+			name: "valid minimal config",
+			data: []byte(`{
+				"version": "v1",
+				"authConfig": {
+					"subscriptionId": "test-subscription"
+				},
+				"clusterConfig": {
+					"resourceGroup": "test-rg",
+					"location": "eastus"
+				},
+				"apiServerConfig": {
+					"apiServerName": "test-api-server"
+				}
+			}`),
+			want: &aksnodeconfigv1.Configuration{
+				Version: "v1",
+				AuthConfig: &aksnodeconfigv1.AuthConfig{
+					SubscriptionId: "test-subscription",
+				},
+				ClusterConfig: &aksnodeconfigv1.ClusterConfig{
+					ResourceGroup: "test-rg",
+					Location:      "eastus",
+				},
+				ApiServerConfig: &aksnodeconfigv1.ApiServerConfig{
+					ApiServerName: "test-api-server",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "empty data",
+			data:    []byte(""),
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name:    "invalid JSON",
+			data:    []byte(`{"version": "v1", invalid}`),
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name: "unknown field should error",
+			data: []byte(`{
+				"version": "v1",
+				"unknownField": "should be ignored",
+				"authConfig": {
+					"subscriptionId": "test-subscription"
+				}
+			}`),
+			want: &aksnodeconfigv1.Configuration{
+				Version: "v1",
+				AuthConfig: &aksnodeconfigv1.AuthConfig{
+					SubscriptionId: "test-subscription",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "valid config from test file",
+			data: func() []byte {
+				data, err := os.ReadFile("../../parser/testdata/test_aksnodeconfig.json")
+				if err != nil {
+					t.Logf("Could not read test file, skipping: %v", err)
+					return []byte(`{"version": "v1"}`)
+				}
+				return data
+			}(),
+			want:    nil, // We'll check for non-nil result instead of exact match
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := UnmarshalConfigurationV1(tt.data)
+
+			// Check error expectation
+			if (err != nil) != tt.wantErr {
+				t.Errorf("UnmarshalConfigurationV1() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			// Early return for error cases
+			if tt.wantErr {
+				return
+			}
+
+			// Handle special test file case
+			if tt.name == "valid config from test file" {
+				validateTestFileResult(t, got)
+				return
+			}
+
+			// Validate normal test cases
+			validateTestResult(t, got, tt.want)
```

Comment:
I think it's better to inline all the logic here and use `assert` library for assertion.

Something like `assert.Equal(t, tt.want, got)` should do the job?

Reply:
updated.